internal/mode/static/nginx/modules/src/httpmatches.js

```diff
@@ -49,32 +48,37 @@ function redirect(r) {
   r.internalRedirect(match.redirectPath);
 }
 
-function extractMatchesFromRequest(r) {
-  if (!r.variables[MATCHES_VARIABLE]) {
-    throw Error(
-      `cannot redirect the request; the variable ${MATCHES_VARIABLE} is not defined on the request object`,
-    );
-  }
-
-  let matches;
-
+function internalRedirect(r, matchList) {
```

Comment:
the internal redirect function is no longer used in NGINX config. Do we need to keep it?

Reply:
it is still used in the config to do the http match redirect. `matchList` only has list of httpMatches